src/global_scheduler/global_scheduler_algorithm.c

```diff
@@ -24,6 +24,7 @@ void handle_local_scheduler_heartbeat(global_scheduler_state *state) {
 
 void handle_new_local_scheduler(global_scheduler_state *state,
                                 db_client_id db_client_id) {
-  local_scheduler local_scheduler = {.id = db_client_id};
+  local_scheduler local_scheduler;
+  local_scheduler.id = db_client_id;
```

Comment:
Ah I see, I didn't realize the field orders were changing often. In that case, if you do want the zero-initialization, then in lieu of the braces you can do `memset(&local_scheduler, 0, sizeof(local_scheduler))` and then just assign to the field like you have here. If you don't, then just keep what you have as is.